src/core/map/level_generator/random.h

```diff
@@ -3,7 +3,7 @@
 #include <random>
 namespace map {
 
-static std::default_random_engine mRand;
+extern std::default_random_engine mRand;
```

Comment:
default_random_engine& getEngine();, impl can return function-static var, that way we avoid static init before main.